_security-plugin/configuration/ldap.md

```diff
@@ -641,3 +640,33 @@ authz:
         rolename: cn
         resolve_nested_roles: true
 ```
+
+### Connection pooling settings
+
+The directory server can maintain a pool of connections at the ready, assigning them when needed and returning them to the pool after a connection is closed. This arrangement can lower demands on the resources used to create connections, improve OpenSearch performance, and reduce load on the server. You can use the settings below to control the way the pooling is carried out. 
```

Comment:
Opensearch is the one maintaining the pool of connections, not the directory server. The security plugin achieves pooling using this library: https://www.ldaptive.org/docs/guide/connections/pooling.html

I'm also only seeing connection pooling in `LDAPAuthenticationBackend2` here: https://github.com/opensearch-project/security/blob/main/src/main/java/com/amazon/dlic/auth/ldap2/LDAPAuthenticationBackend2.java#L71

Diving in deeper now to determine when `LDAPAuthenticationBackend2` is used instead of `LDAPAuthenticationBackend` and will follow-up with another comment when I determine when each is used.

Reply:
In the tests for `LDAPAuthenticationBackend2` it looks like the `type` in the `config.yml` is the fully qualified classname `com.amazon.dlic.auth.ldap2.LDAPAuthenticationBackend2` and I don't see this anywhere in the documentation website. @nibix would you be able to provide more context about the 2 different LDAP authentication backends?

See https://github.com/opensearch-project/security/blob/main/src/test/resources/ldap/config_ldap2.yml#L35

Reply:
Looks like you can also use the short-hand `ldap2` judging by this: https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/securityconf/DynamicConfigModelV7.java#L349-L356 though I don't see any mention of this on the documentation website.

Reply:
As far as I remember, the major difference  of the `ldap2` module is indeed connection pooling. Historically, it was introduced as an experimental module in Search Guard which was supposed to replace the `ldap` module at some point in time. Thus, it is also undocumented so far.

Reply:
Changed this to, "OpenSearch can maintain a pool of connections..."